pentaho/src/main/java/pt/webdetails/cda/dataaccess/PentahoDataAccessUtils.java

```diff
@@ -115,10 +126,57 @@ public KettleTransformationProducer createKettleTransformationProducer( TransFro
     } else {
       relPath = ktrPath;
     }
-
-    return new CdaPentahoKettleTransFromFileProducer( "",
+    Repository repo = null;
+    try {
+      repo = connectToRepository();
+    } catch ( KettleException | ActionExecutionException e ) {
+      logger.warn( "Failed to connect to repository. " + e.getMessage() );
+    }
+    CdaPentahoKettleTransFromFileProducer cdaPentahoKettleTransFromFileProducer
+        = new CdaPentahoKettleTransFromFileProducer( "",
       relPath, query, null, null, connectionInfo.getDefinedArgumentNames(),
       connectionInfo.getDefinedVariableNames() );
+    cdaPentahoKettleTransFromFileProducer.setRepository( repo );
+    return cdaPentahoKettleTransFromFileProducer;
+  }
+
+  protected Repository connectToRepository() throws KettleException, ActionExecutionException {
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+
+    // only load a default enterprise repository. If this option is set, then you cannot load
+    // transformations or jobs from anywhere but the local server.
+    String repositoriesXml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repositories>" //$NON-NLS-1$
```

Comment:
@pedro, we use same code here: https://github.com/pentaho/pentaho-platform/blob/501ff46382086ce56cd83bab28b48c709edf385b/extensions/src/main/java/org/pentaho/platform/plugin/services/exporter/MetaStoreExportUtil.java#L28 and here: https://github.com/pentaho/pentaho-kettle/blob/68cf1c7220ea3386b3f2be9cadc11ae42bb3e12f/plugins/pdi-pur-plugin/src/org/pentaho/di/repository/utils/IRepositoryFactory.java#L132. I'm not sure of how we can reference to repo since we don't have any information about it while constructing KettleTransformationProducer: https://github.com/YuryBY/cda/blob/a864049df3f091da961cde18f81e479525183dae/pentaho/src/main/java/pt/webdetails/cda/dataaccess/PentahoDataAccessUtils.java#L119